my_enums.rb

```diff
@@ -85,28 +228,36 @@ def my_map_three(param = nil)
     my_each do |n|
       if param.nil? && block_given?
         emp_arr << yield(n)
+      elsif !param.nil? && !block_given?
+        emp_arr << param.call(n)
       elsif !param.nil? && block_given?
         emp_arr << param.call(n)
+      else
+        return to_enum
       end
     end
     emp_arr
   end
 
   def my_inject(inivalue = nil, symbol = nil)
     if !inivalue.nil? && !symbol.nil?
-      my_each { |num| inivalue.method(symbol).call(num) }
+      my_each { |num| inivalue = inivalue.method(symbol).call(num) }
       inivalue
     elsif !inivalue.nil? && inivalue.is_a?(Symbol) && symbol.nil?
-      memo, *rem_elements = self
-      rem_elements.my_each { |num| memo = memo.method(inivalue).call(num) }
+      arr = to_a if is_a? Range
+      arr = self unless is_a? Range
+      memo = arr.shift
+      arr.my_each { |num| memo = memo.method(inivalue).call(num) }
       memo
```

Comment:
- [ ] should have working `my_inject` identical and returns the same thing as ruby's [`inject`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_inject.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

- Your method should work with blocks as arguments. Right now, it has a problem when you do something like this: `stored_array.my_inject{ |sum, n| sum + n } ` It returns a different value (I think it's adding the first element twice)

- Your method should work with symbols. Right now, it has a problem when you do something like this: `stored_array.my_inject(:+) ` It returns a different value (I think it's adding the first element twice).

It's very important that you use a stored array to try to reproduce the error and call the method multiple times on the same array, in case you are mutating the original array.